File: packages/ts-client-sdk/src/sdk/models/operations/queryseatsbyid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class QuerySeatsByIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=seatId" })
  seatId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscriptionId" })
  subscriptionId: string;
}

export class QuerySeatsByIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: QuerySeatsByIdPathParams;
}

export class QuerySeatsByIdDefaultApplicationJSONIssues extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QuerySeatsByIdDefaultApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=code" })
  code: string;

  @SpeakeasyMetadata({ data: "json, name=issues", elemType: QuerySeatsByIdDefaultApplicationJSONIssues })
  issues?: QuerySeatsByIdDefaultApplicationJSONIssues[];

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;
}

export class QuerySeatsById200ApplicationJSONOccupant extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;

  @SpeakeasyMetadata({ data: "json, name=seat_id" })
  seatId: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_id" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "json, name=user_id" })
  userId: string;

  @SpeakeasyMetadata({ data: "json, name=user_name" })
  userName: string;
}

export class QuerySeatsById200ApplicationJSONReservation extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;

  @SpeakeasyMetadata({ data: "json, name=invite_url" })
  inviteUrl: string;

  @SpeakeasyMetadata({ data: "json, name=seat_id" })
  seatId: string;

  @SpeakeasyMetadata({ data: "json, name=tenant_id" })
  tenantId: string;

  @SpeakeasyMetadata({ data: "json, name=user_id" })
  userId: string;
}
export enum QuerySeatsById200ApplicationJSONSeatTypeEnum {
    Standard = "standard",
    Limited = "limited"
}
export enum QuerySeatsById200ApplicationJSONSeatingStrategyNameEnum {
    MonthlyActiveUser = "monthly_active_user",
    FirstComeFirstServed = "first_come_first_served"
}

export class QuerySeatsById200ApplicationJSON extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=created_utc" })
  createdUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=expires_utc" })
  expiresUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=occupant" })
  occupant: QuerySeatsById200ApplicationJSONOccupant;

  @SpeakeasyMetadata({ data: "json, name=redeemed_utc" })
  redeemedUtc: Date;

  @SpeakeasyMetadata({ data: "json, name=reservation" })
  reservation: QuerySeatsById200ApplicationJSONReservation;

  @SpeakeasyMetadata({ data: "json, name=seat_type" })
  seatType: QuerySeatsById200ApplicationJSONSeatTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=seating_strategy_name" })
  seatingStrategyName: QuerySeatsById200ApplicationJSONSeatingStrategyNameEnum;

  @SpeakeasyMetadata({ data: "json, name=subscription_id" })
  subscriptionId: string;
}

export class QuerySeatsByIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  querySeatsById200ApplicationJSONObject?: QuerySeatsById200ApplicationJSON;

  @SpeakeasyMetadata()
  querySeatsByIdDefaultApplicationJSONObject?: QuerySeatsByIdDefaultApplicationJSON;
}

File: packages/go-client-sdk/pkg/models/operations/queryseatsbyid.go
package operations

import (
	"time"
)

type QuerySeatsByIDPathParams struct {
	SeatID         string `pathParam:"style=simple,explode=false,name=seatId"`
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscriptionId"`
}

type QuerySeatsByIDRequest struct {
	PathParams QuerySeatsByIDPathParams
}

type QuerySeatsByIDDefaultApplicationJSONIssues struct {
	Message string `json:"message"`
}

type QuerySeatsByIDDefaultApplicationJSON struct {
	Code    string                                       `json:"code"`
	Issues  []QuerySeatsByIDDefaultApplicationJSONIssues `json:"issues,omitempty"`
	Message string                                       `json:"message"`
}

type QuerySeatsByID200ApplicationJSONOccupant struct {
	Email    string `json:"email"`
	SeatID   string `json:"seat_id"`
	TenantID string `json:"tenant_id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

type QuerySeatsByID200ApplicationJSONReservation struct {
	Email     string `json:"email"`
	InviteURL string `json:"invite_url"`
	SeatID    string `json:"seat_id"`
	TenantID  string `json:"tenant_id"`
	UserID    string `json:"user_id"`
}

type QuerySeatsByID200ApplicationJSONSeatTypeEnum string

const (
	QuerySeatsByID200ApplicationJSONSeatTypeEnumStandard QuerySeatsByID200ApplicationJSONSeatTypeEnum = "standard"
	QuerySeatsByID200ApplicationJSONSeatTypeEnumLimited  QuerySeatsByID200ApplicationJSONSeatTypeEnum = "limited"
)

type QuerySeatsByID200ApplicationJSONSeatingStrategyNameEnum string

const (
	QuerySeatsByID200ApplicationJSONSeatingStrategyNameEnumMonthlyActiveUser    QuerySeatsByID200ApplicationJSONSeatingStrategyNameEnum = "monthly_active_user"
	QuerySeatsByID200ApplicationJSONSeatingStrategyNameEnumFirstComeFirstServed QuerySeatsByID200ApplicationJSONSeatingStrategyNameEnum = "first_come_first_served"
)

type QuerySeatsByID200ApplicationJSON struct {
	CreatedUtc          time.Time                                               `json:"created_utc"`
	ExpiresUtc          time.Time                                               `json:"expires_utc"`
	ID                  string                                                  `json:"id"`
	Occupant            QuerySeatsByID200ApplicationJSONOccupant                `json:"occupant"`
	RedeemedUtc         time.Time                                               `json:"redeemed_utc"`
	Reservation         QuerySeatsByID200ApplicationJSONReservation             `json:"reservation"`
	SeatType            QuerySeatsByID200ApplicationJSONSeatTypeEnum            `json:"seat_type"`
	SeatingStrategyName QuerySeatsByID200ApplicationJSONSeatingStrategyNameEnum `json:"seating_strategy_name"`
	SubscriptionID      string                                                  `json:"subscription_id"`
}

type QuerySeatsByIDResponse struct {
	ContentType                                string
	StatusCode                                 int64
	QuerySeatsByID200ApplicationJSONObject     *QuerySeatsByID200ApplicationJSON
	QuerySeatsByIDDefaultApplicationJSONObject *QuerySeatsByIDDefaultApplicationJSON
}
